Main svc

rocketman27 opened the pull request.

rocketman27 committed:
feat: Added stat-service
- Added stats-dto, stats-client, stats-server,
- Added docker-compose.yml, Dockerfiles for main-service and stats-service,
- Added main classes for 2 main modules

rocketman27 committed:
fix: Refactored stat-service
- Renamed packages,
- Updated pom files for patent its submodules,
- Added persistence annotations to RequestInfo

rocketman27 committed:
feat: Added main-service
- Added controllers, services, repositories, models
- Added openapi-codegen to generate dto and api controllers,
- Added jpamodelgen to generate metamodels for specifications,
- Added mapstruct to generate mappers,
- Added schema.sql for main-service

utausheva commented on main-service/pom.xml:
```diff
@@ -0,0 +1,198 @@
+        <dependency>
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-boot-starter</artifactId>
+            <version>3.0.0</version>
```
Те зависимости, которые уже включены в spring boot’овский BOM, должны быть без версий, а если ты используешь то, чего там нет — тогда такая зависимость должна быть прописана в блоке корневого pom’а, где и будет указана версия. А в pom’е самого модуля (где зависимость непосредственно используется) должно быть всё равно без версии. Этим обеспечивается центральное место для управления версиями зависимостей и соответственно порядок в проекте.

rocketman27 replied:
Исправлено

utausheva commented on main-service/src/main/java/ru/practicum/user/model/User.java:
```diff
@@ -0,0 +1,35 @@
+    @Column(name = "id", nullable = false)
+    private Long id;
+
+    @Column(name = "email")
```
давай length тоже перенесем на все поля String всех моделей

rocketman27 replied:
Исправлено

utausheva commented on main-service/src/main/java/ru/practicum/request/repository/ParticipationRequestRepository.java:
```diff
@@ -0,0 +1,20 @@
+
+    Optional<ParticipationRequest> findByIdAndRequesterId(long requestId, long requesterId);
+
+    Optional<ParticipationRequest> findByRequesterIdAndEventId(long requesterId, long eventId);
```
некоторые методы не используются в работе, их можно удалить

rocketman27 replied:
Исправлено

utausheva commented on main-service/src/main/java/ru/practicum/request/service/ParticipationRequestServiceImpl.java:
```diff
@@ -0,0 +1,253 @@
+
+    @Override
+    public List<ParticipationRequestDto> getEventParticipants(Long userId, Long eventId) {
+        User user = userRepository.findById(userId)
```
переменная user не используется, ее можно не объявлять

rocketman27 replied:
Исправлено

utausheva commented on main-service/src/main/java/ru/practicum/request/service/ParticipationRequestServiceImpl.java:
```diff
@@ -0,0 +1,253 @@
+
+@Slf4j
+@Service
+public class ParticipationRequestServiceImpl implements ParticipationRequestService {
```
некоторым методам в этом классе не помешает выполнение внутри транзакции

rocketman27 replied:
Исправлено

utausheva reviewed (commented):
Привет!
Работа вау!! Обычно на дипломе все используют уже изученные библиотеки, темы, подходы, а ты пошел дальше, это очень круто!!
буквально пара комментариев, исправление которых не займет много времени, только нужно проверить все аналогичные классы на замечания (в особенности length и транзакции)

rocketman27 committed:
fix: Fixed issues
- Added length to text fields of the models,
- Removed unused variables and methods,
- Marked some methods as Transactional,
- Moved dependencies to the parent pom file

utausheva reviewed (approved):
Отлично! Можешь мерджить)
